statement_test.go

```diff
@@ -439,3 +440,36 @@ func TestMultiStatementCountMismatch(t *testing.T) {
 		t.Fatal("should have failed to query multiple statements")
 	}
 }
+
+func TestGetQueryID(t *testing.T) {
+	var db *sql.DB
+	var err error
+
+	if db, err = sql.Open("snowflake", dsn); err != nil {
+		t.Fatalf("failed to open db. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	ctx := context.TODO()
+	conn, _ := db.Conn(ctx)
+
+	err = conn.Raw(func(x interface{}) error {
+		stmt, err := x.(driver.ConnPrepareContext).PrepareContext(ctx, "select 1")
+		if err != nil {
+			return err
+		}
+		rows, err := stmt.(driver.StmtQueryContext).QueryContext(ctx, nil)
```

Comment:
This test is good. What will happen if `QueryContext` fails? For example, a SQL syntax error should return an error code along with a query id.

Reply:
If `QueryContext` fails, there will be no associated query ID since the entire query would fail. An example of that would be a SQL syntax, but it doesn't execute in the backend so there is no query ID to speak of.

Reply:
There must be a query_id for failed queries. Otherwise the customers cannot locate which query failed. Can you find out the query id in the response from query-request? It is supported by Python Connection, btw.

Reply:
Oh, I see. It hasn't been done in this driver it looks like - we just return the error immediately. I'll look to see how we can extract that information, but it most likely will bring about a behavior change (we return `nil` at the moment as a result along with an error, but we'll have to actually return an object).

Reply:
So yes, I can return a row/result, but it replaces the existing return value of `nil`. In the case that the customers check the nullity of the returned object instead of the error returned, this will cause an issue.

Reply:
@sfc-gh-stakeda I just pushed a prototype of what it'd look like if we were to return a failed SQL query object with the corresponding query ID.

Reply:
can you include the query id in SnowflakeError?